meson.build

```diff
@@ -14,7 +14,9 @@ add_project_arguments(
     '-Wstrict-prototypes',
     '-Wmissing-prototypes',
     '-Wredundant-decls',
-    '-Wshadow'
+    '-Wshadow',
+    '-Wno-error=discarded-qualifiers',
```

Comment:
I would indeed prefer to fix the warnings rather than just suppress them.

But... when I tried enabling `-Wdiscarded-qualifiers` on my system (Fedora 37, gcc-12.2.1-4.fc37.x86_64), I didn't get any warnings.  What are you seeing with this one?

Reply:
@dgibson actually it triggers when using `-Wwrite-strings`